### PR TITLE
Support s3_data_dir and s3_data_naming

### PR DESCRIPTION
**Pull request by thenaturalist**
Ported PR from https://github.com/Tomme/dbt-athena/pull/50 by @chronitis

### CODE REVIEW

**Comment by mattiamatrix** [comment]
@thenaturalist @nicor88 what is preventing this from behind merged? My team has been waiting for this for a very long time ;)

File: dbt/adapters/athena/impl.py [hunk 1]
```diff
@@ -38,11 +38,64 @@ def convert_datetime_type(
...
+        We generate a value here even if `s3_data_dir` is not set,
+        since creating a seed table requires a non-default location.
+        """
         conn = self.connections.get_thread_connection()
-        client = conn.handle
+        creds = conn.credentials
+        if creds.s3_data_dir is not None:
+            return creds.s3_data_dir
+        else:
+            return f"{creds.s3_staging_dir}tables/"
```
Comment (nicor88) [inline dbt/adapters/athena/impl.py:55]: If `s3_staging_dir ` has a trailing slash or it miss a / we can have issues. We can do simply:
`os.path.join(creds.s3_staging_dir, 'tables/')`

**Comment by nicor88** [comment]
@mattiamatrix we wanted to full test the feature with all supported cases, the implementation looks actually good to me, I left a tiny comment. Also the README needs to update to make a bit more clear the usage.

**Comment by mattiamatrix** [comment]
We might also want to ping the people (@chronitis @fbrueck) who were initially involved if they are still interested.

**Comment by chronitis** [comment]
I originally opened this, and we used it in production for maybe 6 months before we moved away from Athena. So I'm not in a position to test it further now, but if it's useful to anyone else, please feel free to merge.

It was workable and having predictable S3 paths made organising the resulting data much easier, although one issue encountered was that (at the time, I haven't checked if this has now changed) the temporary table used when building an incremental table always had a fixed name (table + `_dbt_tmp`) and hence a fixed path. That meant it wasn't possible to generate two independent sets of partitions at the same time. Fixed in https://github.com/northvolt/dbt-athena/commit/223d2cccde487637d41a4ed2d827716fd0ab8cf9

**Comment by chronitis** [comment]
nb, one other feature I wrote which might be of interest: using the Glue API for listing tables, schemas, columns instead of INFORMATION_SCHEMA queries, which were generally a lot slower. I don't think I ever opened a PR for it, but again, you're welcome to pull it if it looks useful: https://github.com/northvolt/dbt-athena/commit/49c7111eab13182ec27c3b7e11d7453ab005c550

**Comment by nicor88** [comment]
Regarding the not unique path, I also add a workaround in the Iceberg implementation, I think that needs to be revisit a bit, but was similar to what you suggested.

Regarding https://github.com/dbt-athena/dbt-athena/pull/4#issuecomment-1305576711 definitely worth to add it. 

@mattiamatrix and @chronitis I'm wondering if you both want to be added as maintainers to help us out to deliver feature faster, we fork the repo to enable faster merge, but also me and @thenaturalist are quite busy. Feel free to reach out to me in Slack dbt.

**Comment by rumbin** [comment]
@nicor88 could you please provide a hint to why this PR has been closed?

**Comment by nicor88** [comment]
@rumbin just a mistake on when we relesed the 1.0.4, the branch was deleted and the pr closed, we will re-open again.

**Comment by nicor88** [comment]
@mattiamatrix feel free to re-open it.

**Pull request closed without merging**